lib/util/http/throttle.ts

```diff
@@ -0,0 +1,52 @@
+import is from '@sindresorhus/is';
+import pThrottle from 'p-throttle';
+import { logger } from '../../logger';
+import { parseUrl } from '../url';
+import { getThrottleInterval, getThrottleLimit } from './host-rules';
+
+const hostThrottles = new Map<string, Throttle | null>();
+
+class Throttle {
+  constructor(
+    limit: number,
+    interval: number,
+    private throttle = pThrottle({
```

Comment:
does it need to be a constructor arg?

Reply:
This would lead to way more ugly types. I think it's better to rely on inferred types here.

Reply:
i don't see any generic here, so i don't understand what the problem with a static type. is there no definition for the return value?